CheckpointPlugin.cpp

```diff
@@ -630,33 +715,32 @@ bool CheckpointPlugin::rewind(ServerWrapper sw) {
 	rewindState.buttonsDown = buttonsDown;
 
 	// Determine how much to rewind / advance time.
-	if (abs(ci.Steer) < .05f) { // Ignore slight input; keep current game state.
+	if (abs(activeInput) < .05f) { // Ignore slight input; keep current game state.
 		return true; // Ignoring input; apply state.
 	}
 	rewindState.deleting = false;
-	if (ci.Steer < -.95 && rewindState.holdingFor <= 0) {
+	if (activeInput < -.95 && rewindState.holdingFor <= 0) {
 		rewindState.holdingFor -= elapsed;
-	} else if (ci.Steer > .95 && rewindState.holdingFor >= 0) {
+	} else if (activeInput > .95 && rewindState.holdingFor >= 0) {
 		rewindState.holdingFor += elapsed;
 	} else {
 		rewindState.holdingFor = 0;
 	}
-	float factor = std::clamp(abs(rewindState.holdingFor) * 2, 1.0f, 10.0f);
-
-	// How much (in seconds) to move "current" (positive or negative)
-	float deltaElapsed = factor * elapsed * ci.Steer; // full left = 2-5 seconds/second
 
+	float factor = std::clamp(fabs(activeInput) * 2, 1.0f, 10.0f);
+	float deltaElapsed = factor * elapsed * activeInput;
 	rewindState.virtualTimeOffset = std::clamp(
-		rewindState.virtualTimeOffset + deltaElapsed, -snapshotInterval * history.size(), .0f);
+		rewindState.virtualTimeOffset + deltaElapsed, -snapshotInterval * history.size(), 0.0f);
+
 	float historyOffset = rewindState.virtualTimeOffset / snapshotInterval;
-	size_t current = std::clamp<size_t>(
-		history.size() + size_t(floor(historyOffset)), 0, history.size() - 1);
-	if (current < (history.size() - 1) /* && NEED TO INTERPOLATE */) {
+	size_t current = std::clamp<size_t>(history.size() + size_t(floor(historyOffset)), 0, history.size() - 1);
+
+	if (current < history.size() - 1) {
 		float advancePct = 1 - (historyOffset - floor(historyOffset));
-		latest = GameState(history.at(current), history.at(current+1), advancePct);
-		return true; // Apply new state.
+		latest = GameState(history.at(current), history.at(current + 1), advancePct);
+	} else {
+		latest = history.at(current);
 	}
-	latest = history.at(current);
 	return true; // Apply new state.
```

Comment:
I don't _think_ this changes the logic at all?  If that's the case, please revert these diffs.